### ISSUE TITLE
An error occurs during training: NotImplementedError: Cannot convert a symbolic Tensor (filtered_detections / map / while / strided_slice_1: 0) to a numpy array.

### ISSUE THREAD

**Issue by nikonnext** [ybkscht/EfficientPose#27]
Hi @ybkscht 
I am running your project on a new Ubuntu 18.04 system. I install your repository according to the recommendation: `conda create -n myenv python==3.7`-> `conda install tensorflow-gpu==1.15.0`-> `pip install -r requirements.txt`->`python setup.py build_ext --inplace`.

And at startup `python train.py --phi 0 linemod /path_to_dataset/Linemod_preprocessed/ --object-id 1 ` 
I am getting the following output:

```
WARNING:tensorflow:From train.py:204: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.

WARNING:tensorflow:From train.py:206: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

2021-05-28 23:29:35.653675: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX
2021-05-28 23:29:35.677039: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3292420000 Hz
2021-05-28 23:29:35.677415: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55a00b0c02e0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-05-28 23:29:35.677448: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2021-05-28 23:29:35.678934: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2021-05-28 23:29:35.702501: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:29:35.702994: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.62
pciBusID: 0000:01:00.0
2021-05-28 23:29:35.703249: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2021-05-28 23:29:35.704563: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
2021-05-28 23:29:35.705550: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10.0
2021-05-28 23:29:35.705804: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10.0
2021-05-28 23:29:35.707095: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10.0
2021-05-28 23:29:35.708178: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10.0
2021-05-28 23:29:35.711536: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2021-05-28 23:29:35.711733: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:29:35.712140: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:29:35.712444: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2021-05-28 23:29:35.712531: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2021-05-28 23:29:35.795894: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-05-28 23:29:35.795966: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2021-05-28 23:29:35.795976: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2021-05-28 23:29:35.796188: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:29:35.796540: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:29:35.796895: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:29:35.797193: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3237 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)
2021-05-28 23:29:35.798773: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55a00afbe8b0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2021-05-28 23:29:35.798810: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): GeForce GTX 1650, Compute Capability 7.5
{'dataset_type': 'linemod', 'rotation_representation': 'axis_angle', 'weights': None, 'freeze_backbone': False, 'no_freeze_bn': False, 'batch_size': 1, 'lr': 0.0001, 'no_color_augmentation': False, 'no_6dof_augmentation': False, 'phi': 0, 'gpu': None, 'epochs': 500, 'steps': 1790, 'snapshot_path': 'checkpoints/28_05_2021_23_29_35', 'tensorboard_dir': 'logs/28_05_2021_23_29_35', 'snapshots': True, 'evaluation': True, 'compute_val_loss': False, 'score_threshold': 0.5, 'validation_image_save_path': None, 'multiprocessing': False, 'workers': 4, 'max_queue_size': 10, 'linemod_path': 'linemod_preproc', 'object_id': 1}

Creating the Generators...
Done!

Building the Model...
WARNING:tensorflow:From /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/util/deprecation.py:507: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with distribution=normal is deprecated and will be removed in a future version.
Instructions for updating:
`normal` is a deprecated alias for `truncated_normal`
WARNING:tensorflow:From /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/ops/resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
2021-05-28 23:30:00.743118: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.743751: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.62
pciBusID: 0000:01:00.0
2021-05-28 23:30:00.743823: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2021-05-28 23:30:00.743858: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
2021-05-28 23:30:00.743890: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10.0
2021-05-28 23:30:00.743920: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10.0
2021-05-28 23:30:00.743949: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10.0
2021-05-28 23:30:00.743978: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10.0
2021-05-28 23:30:00.744007: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2021-05-28 23:30:00.744103: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.744458: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.744804: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2021-05-28 23:30:00.745209: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.745535: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.62
pciBusID: 0000:01:00.0
2021-05-28 23:30:00.745565: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2021-05-28 23:30:00.745598: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
2021-05-28 23:30:00.745638: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10.0
2021-05-28 23:30:00.745669: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10.0
2021-05-28 23:30:00.745692: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10.0
2021-05-28 23:30:00.745712: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10.0
2021-05-28 23:30:00.745734: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2021-05-28 23:30:00.745816: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.746147: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.746434: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2021-05-28 23:30:00.746465: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-05-28 23:30:00.746476: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2021-05-28 23:30:00.746484: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2021-05-28 23:30:00.746569: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.746902: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:983] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2021-05-28 23:30:00.747202: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3237 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)
Traceback (most recent call last):
  File "train.py", line 368, in <module>
    main()
  File "train.py", line 137, in main
    num_rotation_parameters = num_rotation_parameters)
  File "/home/nikita/EfficientPose/model.py", line 138, in build_EfficientPose
    )([bboxes, classification, rotation, translation])
  File "/home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 854, in __call__
    outputs = call_fn(cast_inputs, *args, **kwargs)
  File "/home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/autograph/impl/api.py", line 237, in wrapper
    raise e.ag_error_metadata.to_exception(e)
NotImplementedError: in converted code:

    /home/nikita/EfficientPose/layers.py:443 _filter_detections  *
        return filter_detections(
    /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/ops/map_fn.py:268 map_fn
        maximum_iterations=n)
    /home/nikita/EfficientPose/layers.py:342 filter_detections  *
        labels = c * tf.ones((keras.backend.shape(scores)[0],), dtype='int64')
    /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/ops/array_ops.py:2560 ones
        output = _constant_if_small(one, shape, dtype, name)
    /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/ops/array_ops.py:2295 _constant_if_small
        if np.prod(shape) < 1000:
    <__array_function__ internals>:6 prod
        
    /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3031 prod
        keepdims=keepdims, initial=initial, where=where)
    /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/numpy/core/fromnumeric.py:87 _wrapreduction
        return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
    /home/nikita/anaconda3/envs/AP/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py:736 __array__
        " array.".format(self.name))

    NotImplementedError: Cannot convert a symbolic Tensor (filtered_detections/map/while/strided_slice_1:0) to a numpy array.

```
As I understand this is a bug with numpy, but I cannot solve it, please help.

My pip list:

```
Package                         Version
------------------------------- -------------------
absl-py                         0.12.0
actionlib                       1.12.1
angles                          1.9.12
arbotix-python                  0.10.0
astor                           0.8.1
base-local-planner              1.16.7
bondpy                          1.8.5
cached-property                 1.5.2
camera-calibration              1.15.0
camera-calibration-parsers      1.11.13
catkin                          0.7.29
certifi                         2020.12.5
cob-cartesian-controller        0.8.12
cob-script-server               0.6.19
cob-twist-controller            0.8.12
controller-manager              0.18.3
controller-manager-msgs         0.18.3
controller-manager-tests        0.18.3
cv-bridge                       1.13.0
cycler                          0.10.0
Cython                          0.29.23
decorator                       4.4.2
diagnostic-analysis             1.9.7
diagnostic-common-diagnostics   1.9.7
diagnostic-updater              1.9.7
dynamic-reconfigure             1.6.3
dynamixel-sdk                   3.7.51
gast                            0.2.2
gazebo-plugins                  2.8.7
gazebo-ros                      2.8.7
gencpp                          0.6.5
geneus                          2.2.6
genlisp                         0.4.16
genmsg                          0.5.16
gennodejs                       2.0.1
genpy                           0.6.16
google-pasta                    0.2.0
grpcio                          1.14.1
h5py                            3.2.1
image-geometry                  1.13.0
imageio                         2.9.0
imgaug                          0.4.0
importlib-metadata              3.10.0
interactive-markers             1.11.5
joint-state-publisher           1.12.15
joint-state-publisher-gui       1.12.15
kdl-parser-py                   1.13.1
Keras-Applications              1.0.8
Keras-Preprocessing             1.1.2
kiwisolver                      1.3.1
laser-geometry                  1.6.7
Markdown                        3.3.4
matplotlib                      3.4.2
message-filters                 1.14.11
mkl-fft                         1.3.0
mkl-random                      1.2.1
mkl-service                     2.3.0
moveit-commander                1.0.7
moveit-python                   0.4.1
moveit-ros-planning-interface   1.0.7
moveit-ros-visualization        1.0.7
networkx                        2.5.1
numpy                           1.20.3
opencv-python                   4.5.2.52
opt-einsum                      3.3.0
pandas                          1.1.5
Pillow                          8.2.0
pip                             21.1.1
plyfile                         0.7.4
pr2-controller-manager          1.8.18
progressbar2                    3.53.1
protobuf                        3.14.0
pyparsing                       2.4.7
python-dateutil                 2.8.1
python-qt-binding               0.4.3
python-utils                    2.5.6
pytz                            2021.1
PyWavelets                      1.1.1
PyYAML                          5.4.1
qt-dotgraph                     0.4.2
qt-gui                          0.4.2
qt-gui-cpp                      0.4.2
qt-gui-py-common                0.4.2
resource-retriever              1.12.6
rosbag                          1.14.11
rosboost-cfg                    1.14.9
rosclean                        1.14.9
roscreate                       1.14.9
rosgraph                        1.14.11
roslaunch                       1.14.11
roslib                          1.14.9
roslint                         0.11.2
roslz4                          1.14.11
rosmake                         1.14.9
rosmaster                       1.14.11
rosmsg                          1.14.11
rosnode                         1.14.11
rosparam                        1.14.11
rospy                           1.14.11
rosservice                      1.14.11
rostest                         1.14.11
rostopic                        1.14.11
rosunit                         1.14.9
roswtf                          1.14.11
rqt-action                      0.4.9
rqt-bag                         0.5.1
rqt-bag-plugins                 0.5.1
rqt-console                     0.4.9
rqt-controller-manager          0.18.3
rqt-dep                         0.4.9
rqt-graph                       0.4.11
rqt-gui                         0.5.2
rqt-gui-py                      0.5.2
rqt-image-view                  0.4.16
rqt-joint-trajectory-controller 0.17.2
rqt-launch                      0.4.8
rqt-logger-level                0.4.8
rqt-moveit                      0.5.10
rqt-msg                         0.4.8
rqt-nav-view                    0.5.7
rqt-plot                        0.4.13
rqt-pose-view                   0.5.8
rqt-publisher                   0.4.8
rqt-py-common                   0.5.2
rqt-py-console                  0.4.8
rqt-reconfigure                 0.5.4
rqt-robot-dashboard             0.5.7
rqt-robot-monitor               0.5.13
rqt-robot-steering              0.5.10
rqt-runtime-monitor             0.5.7
rqt-rviz                        0.6.0
rqt-service-caller              0.4.8
rqt-shell                       0.4.9
rqt-srv                         0.4.8
rqt-tf-tree                     0.6.0
rqt-top                         0.4.8
rqt-topic                       0.4.11
rqt-web                         0.4.8
rviz                            1.13.17
scikit-image                    0.18.1
scipy                           1.6.2
seed-smartactuator-sdk          0.0.5
sensor-msgs                     1.12.8
setuptools                      52.0.0.post20210125
Shapely                         1.7.1
six                             1.16.0
smach                           2.0.1
smach-ros                       2.0.1
smclib                          1.8.5
srdfdom                         0.5.2
tensorboard                     1.15.0
tensorflow                      1.15.0
tensorflow-estimator            1.15.1
termcolor                       1.1.0
tf                              1.12.1
tf-conversions                  1.12.1
tf2-geometry-msgs               0.6.5
tf2-kdl                         0.6.5
tf2-py                          0.6.5
tf2-ros                         0.6.5
tifffile                        2021.4.8
topic-tools                     1.14.11
tqdm                            4.61.0
typeguard                       2.12.0
typing-extensions               3.7.4.3
urdfdom-py                      0.4.5
webencodings                    0.5.1
Werkzeug                        0.16.1
wheel                           0.36.2
wrapt                           1.12.1
xacro                           1.13.10
zipp                            3.4.1
```
my conda list:

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_tflow_select             2.1.0                       gpu  
absl-py                   0.12.0           py37h06a4308_0  
astor                     0.8.1            py37h06a4308_0  
binutils_impl_linux-64    2.31.1               h6176602_1  
binutils_linux-64         2.31.1               h6176602_9  
blas                      1.0                         mkl  
c-ares                    1.17.1               h27cfd23_0  
ca-certificates           2021.4.13            h06a4308_1  
cached-property           1.5.2                      py_0  
certifi                   2020.12.5        py37h06a4308_0  
cudatoolkit               10.0.130                      0  
cudnn                     7.6.5                cuda10.0_0  
cupti                     10.0.130                      0  
cycler                    0.10.0                   pypi_0    pypi
cython                    0.29.23                  pypi_0    pypi
decorator                 4.4.2                    pypi_0    pypi
gast                      0.2.2                    py37_0  
gcc_impl_linux-64         7.3.0                habb00fd_1  
gcc_linux-64              7.3.0                h553295d_9  
google-pasta              0.2.0                      py_0  
grpcio                    1.14.1           py37h9ba97e2_0  
gxx_impl_linux-64         7.3.0                hdf63c60_1  
gxx_linux-64              7.3.0                h553295d_9  
h5py                      3.2.1            py37h6c542dc_0  
hdf5                      1.10.6               hb1b8bf9_0  
imageio                   2.9.0                    pypi_0    pypi
imgaug                    0.4.0                    pypi_0    pypi
importlib-metadata        3.10.0           py37h06a4308_0  
intel-openmp              2021.2.0           h06a4308_610  
keras-applications        1.0.8                      py_1  
keras-preprocessing       1.1.2              pyhd3eb1b0_0  
kiwisolver                1.3.1                    pypi_0    pypi
libedit                   3.1.20210216         h27cfd23_1  
libffi                    3.2.1             hf484d3e_1007  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libprotobuf               3.14.0               h8c45485_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
markdown                  3.3.4            py37h06a4308_0  
matplotlib                3.4.2                    pypi_0    pypi
mkl                       2021.2.0           h06a4308_296  
mkl-service               2.3.0            py37h27cfd23_1  
mkl_fft                   1.3.0            py37h42c9631_2  
mkl_random                1.2.1            py37ha9443f7_2  
ncurses                   6.2                  he6710b0_1  
networkx                  2.5.1                    pypi_0    pypi
numpy                     1.20.3                   pypi_0    pypi
opencv-python             4.5.2.52                 pypi_0    pypi
openssl                   1.0.2u               h7b6447c_0  
opt_einsum                3.3.0              pyhd3eb1b0_1  
pandas                    1.1.5                    pypi_0    pypi
pillow                    8.2.0                    pypi_0    pypi
pip                       21.1.1           py37h06a4308_0  
plyfile                   0.7.4                    pypi_0    pypi
progressbar2              3.53.1                   pypi_0    pypi
protobuf                  3.14.0           py37h2531618_1  
pyparsing                 2.4.7                    pypi_0    pypi
python                    3.7.0                h6e4f718_3  
python-dateutil           2.8.1                    pypi_0    pypi
python-utils              2.5.6                    pypi_0    pypi
pytz                      2021.1                   pypi_0    pypi
pywavelets                1.1.1                    pypi_0    pypi
pyyaml                    5.4.1                    pypi_0    pypi
readline                  7.0                  h7b6447c_5  
scikit-image              0.18.1                   pypi_0    pypi
scipy                     1.6.2            py37had2a1c9_1  
setuptools                52.0.0           py37h06a4308_0  
shapely                   1.7.1                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
sqlite                    3.33.0               h62c20be_0  
tensorboard               1.15.0             pyhb230dea_0  
tensorflow                1.15.0          gpu_py37h0f0df58_0  
tensorflow-base           1.15.0          gpu_py37h9dcbed7_0  
tensorflow-estimator      1.15.1             pyh2649769_0  
tensorflow-gpu            1.15.0               h0d30ee6_0  
termcolor                 1.1.0            py37h06a4308_1  
tifffile                  2021.4.8                 pypi_0    pypi
tk                        8.6.10               hbc83047_0  
tqdm                      4.61.0                   pypi_0    pypi
typeguard                 2.12.0                   pypi_0    pypi
typing_extensions         3.7.4.3            pyha847dfd_0  
webencodings              0.5.1                    py37_1  
werkzeug                  0.16.1                     py_0  
wheel                     0.36.2             pyhd3eb1b0_0  
wrapt                     1.12.1           py37h7b6447c_1  
xz                        5.2.5                h7b6447c_0  
zipp                      3.4.1              pyhd3eb1b0_0  
zlib                      1.2.11               h7b6447c_3  

```

**Comment by ybkscht** [comment]
Hi @nikonnext ,

The same issue is already solved [here](https://github.com/ybkscht/EfficientPose/issues/16). 
The solution is to downgrade numpy. I tested version 1.18.5 but according to this [issue](https://github.com/ybkscht/EfficientPose/issues/17) 1.19.5 seems to be fine too.

Sincerely,
Yannick

**Comment by nikonnext** [comment]
@ybkscht ,
Thanks for a possible solution, however I tried to downgrade numpy to 1.19.5 and 1.18.5. This throws another error:
```
Traceback (most recent call last):
  File "train.py", line 49, in <module>
    from model import build_EfficientPose
  File "/home/nikita/EfficientPose/model.py", line 51, in <module>
    from utils.anchors import anchors_for_shape
  File "/home/nikita/EfficientPose/utils/anchors.py", line 29, in <module>
    from utils.compute_overlap import compute_overlap
  File "utils/compute_overlap.pyx", line 1, in init utils.compute_overlap
    """
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject
```

**Comment by hans24o** [comment]
@nikonnext, you simply have to recompile your compute_overlap with the older numpy version installed as described [here](https://github.com/ybkscht/EfficientPose/issues/16#issuecomment-826663121).

**Comment by nikonnext** [comment]
Thanks @hans24o
 It really helped me. However, a new error has appeared:
 ```
original_keras_version = f.attrs['keras_version'].decode('utf8')
AttributeError: 'str' object has no attribute 'decode'
```
I fixed it with `pip install 'h5py<3.0.0'`
And now everything works!
